samples/Azure.Management.Storage/Generated/BlobContainerData.cs

```diff
@@ -14,7 +14,7 @@
 namespace Azure.Management.Storage
 {
     /// <summary> A class representing the BlobContainer data model. </summary>
-    public partial class BlobContainerData : AzureEntityResource
+    public partial class BlobContainerData : ResourceData
```

Comment:
In this case, `AzureEntityResource` has one more property (etag) than `ResourceData`. Just want to double check that we are fine with such inheritance changes? [AzureEntityResource](https://github.com/Azure/azure-rest-api-specs/blob/33753f31627bd92d2473e4ce49a3656e392d2a47/specification/common-types/resource-management/v3/types.json#L38-L55) is actually a common type (storage refers it from commom type as well), but we suppressed it in ResourceManager. Assuming we want to add it back, then this would inherit from `AzureEntityResource` again, which seems whimsical to me. Are we able to change the behavior to only replace the original base type when the original one has less properties than (or a subset of ) the common type?

Reply:
`AzureEntityResource` is suppressed as a common type to avoid too many resource base types and keep the resource inheritance relatively simple. Here the change actually lines with that purpose.